pkg/daemon/ceph/osd/device.go

```diff
@@ -438,16 +439,46 @@ func getStoreSettings(cfg *osdConfig) (map[string]string, error) {
 	return settings, nil
 }
 
-func writeConfigFile(cfg *osdConfig, context *clusterd.Context, cluster *mon.ClusterInfo, location string, prepareOnly bool) error {
+func WriteConfigFile(context *clusterd.Context, cluster *mon.ClusterInfo, kv *k8sutil.ConfigMapKVStore, osdID int, storeConfig config.StoreConfig, nodeName, location string) error {
+	scheme, err := config.LoadScheme(kv, config.GetConfigStoreName(nodeName))
+	if err != nil {
+		return fmt.Errorf("failed to load partition scheme: %+v", err)
+	}
+
+	config := &osdConfig{id: osdID, configRoot: context.ConfigDir, rootPath: getOSDRootDir(context.ConfigDir, osdID),
+		dir: true, storeConfig: storeConfig, kv: kv, storeName: config.GetConfigStoreName(nodeName)}
+
+	// if a device, search the osd scheme for the requested osd id
+	for _, entry := range scheme.Entries {
+		if entry.ID == osdID {
+			config.partitionScheme = entry
+			config.uuid = entry.OsdUUID
+			config.dir = false
+			break
+		}
+	}
+
+	logger.Infof("updating config for osd %d", osdID)
+	err = writeConfigFile(config, context, cluster, location, false)
+	if err != nil {
+		return err
+	}
+	confFile := getOSDConfFilePath(config.rootPath, cluster.Name)
+	util.WriteFileToLog(logger, confFile)
+	return nil
+}
+
+func writeConfigFile(cfg *osdConfig, context *clusterd.Context, cluster *mon.ClusterInfo, location string, provisioning bool) error {
 	cephConfig := mon.CreateDefaultCephConfig(context, cluster, cfg.rootPath)
 	if isBluestore(cfg) {
 		cephConfig.GlobalConfig.OsdObjectStore = config.Bluestore
 	} else {
 		cephConfig.GlobalConfig.OsdObjectStore = config.Filestore
 	}
 	cephConfig.CrushLocation = location
-	// don't set public or cluster addr if prepare only
-	if prepareOnly {
+
+	// don't set public or cluster addr during provisioning
```

Comment:
why not? this sounds like the same answer to @galexrt question in https://github.com/rook/rook/pull/1945/files#r205942827